presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java

```diff
@@ -641,6 +642,11 @@ public HiveSessionProperties(HiveClientConfig hiveClientConfig, OrcFileWriterCon
                         AFFINITY_SCHEDULING_FILE_SECTION_SIZE,
                         "Size of file section for affinity scheduling",
                         hiveClientConfig.getAffinitySchedulingFileSectionSize(),
+                        false),
+                booleanProperty(
+                        LEGACY_TIMESTAMP_BUCKETING,
+                        "Use legacy bucketing algorithm for timestamp type for bucketed table",
+                        false,
```

Comment:
Make a configuration property for this in HiveClientConfig and set the default value for this session property from there.